addons/manager.go

```diff
@@ -99,29 +100,32 @@ func (o *AddonAgentOptions) AddFlags(cmd *cobra.Command) {
 	flags.StringVar(&o.HubKubeconfigFile, "hub-kubeconfig", o.HubKubeconfigFile, "Location of kubeconfig file to connect to hub cluster.")
 	flags.StringVar(&o.SpokeClusterName, "cluster-name", o.SpokeClusterName, "Name of spoke cluster.")
 	flags.StringVar(&o.DRMode, "mode", o.DRMode, "The DR mode of token exchange addon. Valid values are: 'sync', 'async'")
-	gfs := flag.NewFlagSet("", flag.ExitOnError)
-	o.ZapOpts = zap.Options{
-		Development: true,
-	}
-	o.ZapOpts.BindFlags(gfs)
-	flags.AddGoFlagSet(gfs)
+	flags.BoolVar(&o.DevMode, "dev", false, "Set to true for dev environment (Text logging)")
 }
 
 // RunAgent starts the controllers on agent to process work from hub.
 func (o *AddonAgentOptions) RunAgent(ctx context.Context) {
-	klog.Infof("Starting addon agents.")
-
+	ctrl.SetLogger(zap.New(zap.UseDevMode(o.DevMode)))
```

Comment:
```suggestion
        zapLogger := zap.New(zap.UseDevMode(o.DevMode))
        defer zapLogger.Sync()
	ctrl.SetLogger(zapLogger)
```
Sync() call is important to get all the logs before exiting.

Reply:
This does not seem to work as zap.New(...) returns a logr.Logger. There is no Sync for that .

If i use zapLogger = zap.Must(zap.NewProduction()) , then it returns a *zap.Logger which has ability to sync but then it cannot be typecasted to logr.Logger for setLogger()

Reply:
I have used zapr to convert zap to logr instance.
Should work as expected now